src/test/java/space/obminyashka/items_exchange/end2end/AdvertisementFlowTest.java

```diff
@@ -193,23 +191,57 @@ void getAdvertisement_shouldReturnAdvertisementIfExists() throws Exception {
+                .andExpect(jsonPath("$.content.length()").value("3"));
+    }
+
+    @Test
+    @WithMockUser(username = "admin")
+    @DataSet("database_init.yml")
+    void findAdvertisementBySearchParameters_shouldReturnAdvertisementsByAllParameters() throws Exception {
+        sendUriAndGetResultAction(get(ADV_FILTER)
+                .queryParam("subcategorySearchRequest.categoryId", "1")
+                .queryParam("subcategorySearchRequest.subcategoriesIdValues", "1")
+                .queryParam("advertisementFilter.locationId", "2c5467f3-b7ee-48b1-9451-7028255b757b")
+                .queryParam("advertisementFilter.gender", Gender.FEMALE.name())
+                .queryParam("advertisementFilter.age", AgeRange.FROM_10_TO_12.getValue(), AgeRange.FROM_6_TO_9.getValue())
+                .queryParam("advertisementFilter.clothingSizes", Size.Clothing.FIFTY_SEVEN_2_SIXTY_TWO.getRange())
+                .queryParam("advertisementFilter.season", Season.SUMMER.name(), Season.WINTER.name()), status().isOk())
+                .andExpect(jsonPath("$.content.length()").value("2"));
+    }
+
+    @Test
+    @WithMockUser(username = "admin")
+    @DataSet("database_init.yml")
+    void findAdvertisementBySearchParameters_shouldBeThrownValidationException_WhenSizeFromIncorrectSubcategoryClothes() throws Exception {
+        String subcategoryId = "1";
+        String categoryId = "1";
+        String sizeShoes = String.valueOf(Size.Shoes.ELEVEN_POINT_FIVE.getLength());
+        String sizeClothing = Size.Clothing.FIFTY_SEVEN_2_SIXTY_TWO.getRange();
+
+        MvcResult mvcResult = sendUriAndGetMvcResult(get(ADV_FILTER)
+                .queryParam("subcategorySearchRequest.categoryId", categoryId)
+                .queryParam("subcategorySearchRequest.subcategoriesIdValues", subcategoryId)
+                .queryParam("advertisementFilter.shoesSizes", sizeShoes)
+                .queryParam("advertisementFilter.clothingSizes", sizeClothing), status().isBadRequest());
+        assertTrue(mvcResult.getResponse().getContentAsString().contains("You cannot specify sizes for different categories at the same time"));
+    }
+
+    @Test
+    @WithMockUser(username = "admin")
+    @DataSet("database_init.yml")
+    void findAdvertisementBySearchParameters_shouldBeThrownBadRequestException_WhenSubcategoryNotBelongCategory() throws Exception {
+        String categoryId = "2";
+        String sizeShoes = String.valueOf(Size.Shoes.ELEVEN_POINT_FIVE.getLength());
+        MvcResult mvcResult = sendUriAndGetMvcResult(get(ADV_FILTER)
+                .queryParam("subcategorySearchRequest.categoryId", categoryId)
+                .queryParam("subcategorySearchRequest.subcategoriesIdValues", "14", "1", "3")
+                .queryParam("advertisementFilter.shoesSizes", sizeShoes), status().isBadRequest());
+        assertTrue(mvcResult.getResponse().getContentAsString().contains("These subcategories do not belong to the given category"));
```

Comment:
Assertions.assertThat can provide you with more features for validation of the exception type and message.